questionnaires/tests.py

```diff
@@ -723,3 +728,577 @@ def test_questionnaires_detail_page_permission(self):
 
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(response.data, [])
+
+
+class QuestionnairesTests(TestCase):
```

Comment:
The class name is not specific enough. I think we're testing features of the Admin UI here, so the name should reflect that.

There is also a lot of boilerplate and duplicated code, which makes it harder to understand what is happening in each test. Please tidy up this class to extract common code into utility functions/methods/classes if it makes sense.

Reply:
done